### PR TITLE
To verify kdump on RHEL.

### PR DESCRIPTION
**Pull request by Pavithra1602**
This test triggers kdump on the machine configured in config.yaml and checks if vmcore is generated.

Signed-off-by: pavrampu@linux.vnet.ibm.com <pavrampu@in.ibm.com>

### CODE REVIEW

File: generic/kdump.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class KDUMP(Test):
+
+    def run_cmd_out(self, cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True).strip()
+
+    def setUp(self):
+        sm = SoftwareManager()
+        if not sm.check_installed("openssh*") and not sm.install("openssh*"):
+            self.error("Fail to install openssh required for this test.")
```
Comment (abdhaleegit) [inline generic/kdump.py]: Make sure 'kdump-tools' are installed

File: generic/kdump.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        log_file = os.path.join(self.srcdir, "file")
+        session0 = remote.RemoteRunner("ssh", ip, 22, user_name, password, prompt, "\n", log_file, 100, 10, None)
+        session0.run("cat /etc/os-release", 600, "True")
+        if self.run_cmd_out("cat %s | grep ID | head -1 | cut -d'\"' -f2" % log_file) != "rhel":
+            self.skip("Currently Test is supported only on RHEL")
+        session0.run("kdumpctl status", 600, "True")
+        if self.run_cmd_out("cat %s | grep -Eai 'Kdump is not operational'" % log_file):
+            self.fail("Kdump is not operational")
+        else:
+            self.log.info("Kdump status is operational")
```
Comment (abdhaleegit) [inline generic/kdump.py]: use command systemctl <keyword> kdump.service

File: generic/kdump.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        session1 = remote.remote_login("ssh", ip, 22, user_name, password, prompt, "\n", None, 100, None, None, False)
+        session1.sendline('echo "c" > /proc/sysrq-trigger;')
```
Comment (abdhaleegit) [inline generic/kdump.py]: echo 1 > /proc/sys/kernel/sysrq   before above command

File: generic/kdump.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        time.sleep(600)
+        self.log.info("Connecting after reboot")
+        session2 = remote.RemoteRunner("ssh", ip, 22, user_name, password, prompt, "\n", log_file, 100, 10, None)
+        session2.run("ls -lrt /var/crash", 100, "True")
```
Comment (abdhaleegit) [inline generic/kdump.py]: search for specific files, like vmcore-dmesg.txt, vmcore

File: generic/kdump.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
```
Comment (abdhaleegit) [inline generic/kdump.py]: replace with 2017

File: generic/kdump.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
```
Comment (abdhaleegit) [inline generic/kdump.py]: you can still improve the testcase by adding different kdump setup like:
sed -i \"/GRUB_CMDLINE_LINUX/s/crashkernel=auto/crashkernel=2G-4G:320M,4G-32G:512M,32G-64G:1024M,64G-128G:2048M,128G-:4096M/g\" /etc/default/grub
grub2-mkconfig -o /boot/grub2/grub.cfg

for different values.

Also add support for ubuntu kdump

**Comment by Pavithra1602** [comment]
Raised new PR

**Pull request closed without merging**